sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventData.java

```diff
@@ -279,99 +321,432 @@ public EventData addContext(String key, Object value) {
+     */
+    public String getSessionId() {
+        return amqpAnnotatedMessage.getProperties().getGroupId();
+    }
+
+    /**
+     * Sets the session identifier for a session-aware entity.
+     *
+     * @param sessionId The session identifier to be set.
+     *
+     * @return The updated {@link EventData}.
+     * @throws IllegalArgumentException if {@code sessionId} is too long or if the {@code sessionId} does not match
+     *     the {@code partitionKey}.
+     */
+    public EventData setSessionId(String sessionId) {
+        checkIdLength("sessionId", sessionId, MAX_SESSION_ID_LENGTH);
+        checkSessionId(sessionId);
 
-        SystemProperties(final Map<String, Object> map) {
-            super(map);
-            this.partitionKey = removeSystemProperty(PARTITION_KEY_ANNOTATION_NAME.getValue());
-
-            final String offset = removeSystemProperty(OFFSET_ANNOTATION_NAME.getValue());
-            if (offset == null) {
-                throw new IllegalStateException(String.format(Locale.US,
-                    "offset: %s should always be in map.", OFFSET_ANNOTATION_NAME.getValue()));
-            }
-            this.offset = Long.valueOf(offset);
-            put(OFFSET_ANNOTATION_NAME.getValue(), this.offset);
-
-            final Date enqueuedTimeValue = removeSystemProperty(ENQUEUED_TIME_UTC_ANNOTATION_NAME.getValue());
-            if (enqueuedTimeValue == null) {
-                throw new IllegalStateException(String.format(Locale.US,
-                    "enqueuedTime: %s should always be in map.", ENQUEUED_TIME_UTC_ANNOTATION_NAME.getValue()));
-            }
-            this.enqueuedTime = enqueuedTimeValue.toInstant();
-            put(ENQUEUED_TIME_UTC_ANNOTATION_NAME.getValue(), this.enqueuedTime);
-
-            final Long sequenceNumber = removeSystemProperty(SEQUENCE_NUMBER_ANNOTATION_NAME.getValue());
-            if (sequenceNumber == null) {
-                throw new IllegalStateException(String.format(Locale.US,
-                    "sequenceNumber: %s should always be in map.", SEQUENCE_NUMBER_ANNOTATION_NAME.getValue()));
-            }
-            this.sequenceNumber = sequenceNumber;
-            put(SEQUENCE_NUMBER_ANNOTATION_NAME.getValue(), this.sequenceNumber);
+        amqpAnnotatedMessage.getProperties().setGroupId(sessionId);
+        return this;
+    }
+
+    /**
+     * Gets the scheduled enqueue time of this message.
+     * <p>
+     * This value is used for delayed message availability. The message is safely added to the queue, but is not
+     * considered active and therefore not retrievable until the scheduled enqueue time. Mind that the message may not
+     * be activated (enqueued) at the exact given datetime; the actual activation time depends on the queue's workload
+     * and its state.
+     * </p>
+     *
+     * @return the datetime at which the message will be enqueued in Azure Service Bus
+     */
+    public OffsetDateTime getScheduledEnqueueTime() {
```

Comment:
EH has no scheduled thing

Reply:
According to your comment, remove unnecessary parameters, variables and methods in the new version